### ISSUE TITLE
Cannot export project to Macos

### ISSUE THREAD

**Issue by cdfq152313** [godotengine/godot#50376]
### Godot version

3.3.2.stable.mono

### System information

macOS Big Sur 11.4

### Issue description

I use mono version to export project to macos. When I run the export file (project.app), it crash immediately.
I can run project in Godot IDE(Click "Play"). I also can export project to android/windows and run it without crash.
Only macos export project will cause crash.

### Steps to reproduce

1. Create launch.tscn.
2. Attach launch.cs.
3. Config main scene as launch.tscn.
4. Export to Mac OSX. (Choose "Export Project" to create .dmg)
5. Open dmg file and run app.
6. Crash.

Notice: If I don't create .cs file. The export app will not crash.

Here are crash report. Hope it help.

```
Mono: Log file is: '/Users/denny/Library/Application Support/Godot/app_userdata/mono-crash/mono/mono_logs/2021-07-12_01.41.55_99563.log'
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007fff205afd7d _sigtramp + 29
[2] 2   ???                                 0x0000000000000000 0x0 + 0
[3] 3   mono-crash                          0x0000000106d3a39c mono-crash + 988060
[4] 4   mono-crash                          0x0000000106cf5507 mono-crash + 705799
[5] 5   mono-crash                          0x0000000106d3345a mono-crash + 959578
[6] 6   mono-crash                          0x0000000106c9512e mono-crash + 311598
[7] Object* ClassDB::creator<_GodotSharp>()
[8] Object* ClassDB::creator<MobileVRInterface>()
[9] Object* ClassDB::creator<Expression>()
[10] DirAccess* DirAccess::_create_builtin<DirAccessOSX>()
[11] DirAccess* DirAccess::_create_builtin<DirAccessOSX>()
[12] DirAccess* DirAccess::_create_builtin<DirAccessOSX>()
[13] 13  libdyld.dylib                       0x00007fff20585f5d start + 1
-- END OF BACKTRACE --
[1]    99563 abort      /Volumes/mono-crash/mono-crash.app/Contents/MacOS/mono-crash
Saving session...completed.
```

### Minimal reproduction project

https://github.com/cdfq152313/mono-crash

**Comment by Calinou** [comment]
Are you using an x86-based Mac or an ARM-based Mac?

**Comment by bruvzg** [comment]
> When I run the export file (project.app), it crash immediately.
I can run project in Godot IDE(Click "Play"). 

Do you have "JIT", "Unsigned executable memory" and "DYLD environment variables entitlements" selected in the export dialog?

<img width="397" alt="Screenshot 2021-07-09 at 20 58 55" src="https://user-images.githubusercontent.com/7645683/125206282-268e6900-e28f-11eb-9b2c-989cd1751da3.png">

**Comment by cdfq152313** [comment]
@bruvzg Thank you!!! I can run it now.
BTW, Is there any document mention it? I would check if I miss other thing 😭.

**Comment by bruvzg** [comment]
> BTW, Is there any document mention it? I would check if I miss other thing 😭.

It's not documented. In the `4.x` and `3.4` these entitlements are automatically enabled.

**Comment by cdfq152313** [comment]
> It's not documented. In the `4.x` and `3.4` these entitlements are automatically enabled.
I see. Anyway, thank you so much. You save me a lot of time.

**Comment by bruvzg** [comment]
This should be fixed by #50317 (required entitlements are always enabled on in the Mono builds).
